### PR TITLE
grpc: 1.47.0 -> 1.48.0

### PR DESCRIPTION
**Pull request by marsam**
###### Description of changes

https://github.com/grpc/grpc/releases/tag/v1.48.0

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by marsam** [comment]
@GrahamcOfBorg build bear sysdig libtensorflow

**Comment by marsam** [comment]
@GrahamcOfBorg eval

File: pkgs/development/libraries/grpc/default.nix [hunk 1]
```diff
@@ -20,15 +20,15 @@
 
 stdenv.mkDerivation rec {
   pname = "grpc";
-  version = "1.47.0"; # N.B: if you change this, please update:
+  version = "1.48.0"; # N.B: if you change this, please update:
```
Comment (smancill) [inline pkgs/development/libraries/grpc/default.nix:23]: The comment says that those two python packages should also be updated 👇

File: pkgs/development/libraries/grpc/default.nix [hunk 2]
```diff
@@ -20,15 +20,15 @@
...
     # pythonPackages.grpcio-tools
     # pythonPackages.grpcio-status
 
   src = fetchFromGitHub {
     owner = "grpc";
     repo = "grpc";
     rev = "v${version}";
-    sha256 = "sha256-fMYAos0gQelFMPkpR0DdKr4wPX+nhZSSqeaU4URqgto=";
+    sha256 = "sha256-cR+K3po/9XpYWe+sRXGwzvNAPChrWzYu5D4ygBTKKIQ=";
```
Comment (smancill) [inline pkgs/development/libraries/grpc/default.nix:31]:
```suggestion
    hash = "sha256-cR+K3po/9XpYWe+sRXGwzvNAPChrWzYu5D4ygBTKKIQ=";
```

**Review by smancill — CHANGES_REQUESTED** [review]

File: pkgs/development/libraries/grpc/default.nix (hunk 1 above)
Comment (marsam) [reply pkgs/development/libraries/grpc/default.nix:23]: unfortunately, there are no releases for `v1.48.0` https://pypi.org/project/grpcio-tools/ https://pypi.org/project/grpcio-status/

File: pkgs/development/libraries/grpc/default.nix (hunk 2 above)
Comment (marsam) [reply pkgs/development/libraries/grpc/default.nix:31]: fixed, thanks!

File: pkgs/development/libraries/grpc/default.nix (hunk 1 above)
Comment (smancill) [reply pkgs/development/libraries/grpc/default.nix:23]: I am not really sure then that this should be merged. It looks like we need to wait until the downstream Python projects are updated.

@jonringer was the last one to update that comment in de9249f9dac6a824793b5dd8d19c46c5e53f2422.

Comment (smancill) [reply pkgs/development/libraries/grpc/default.nix:23]: > unfortunately, there are no releases for `v1.48.0` https://pypi.org/project/grpcio-tools/ https://pypi.org/project/grpcio-status/

They have been updated now.

Comment (marsam) [reply pkgs/development/libraries/grpc/default.nix:23]: updated. Thanks!

**Comment by smancill** [comment]
@ofborg build google-cloud-cpp

**Review by smancill — APPROVED** [review]

**Comment by vcunat** [comment]
This broke some tests in `arrow-cpp` (and transitively relatively many package builds).  I don't know why.  https://hydra.nixos.org/build/185903212

**Comment by samuela** [comment]
Created https://github.com/NixOS/nixpkgs/issues/185082 prior to seeing @vcunat's comments.

@marsam Could you submit a fix for arrow-cpp or revert this PR? In the future, please test all PRs with nixpkgs-review prior to merging.

**Comment by cpcloud** [comment]
I'm working on updating arrow-cpp to 9.0.0 and it looks like the tests pass with grpc 1.48.0. I'm still trying to work out some new dependency discovery failures, but we should be able avoid a revert here.

**Comment by cpcloud** [comment]
https://github.com/NixOS/nixpkgs/pull/185124

**Comment by SuperSandro2000** [comment]
First of all we should consider adding arrow-cpp to passthru tests of grpc if it easily breaks by updates and is an important consumer.

Also the changelog does not mention any breaking changes I could find.

> Could you submit a fix for arrow-cpp or revert this PR?

First of all: We are not going to start to hastily revert update because of single downstream failures. The fallout must be big/catastrophically or fixes require unjustifiable amounts of work on our side. If we can fix it by adding a patch from upstream then this is the preferred solution. In any way we should consider multiple solutions before reverting. If that means a package does not build for 5 days then so be it, that's okay and we will manage. master is not a branch where no (temporary) breakages are allowed to exist.

When updating packages that are dependencies of many other packages then we must be careful if there are breaking changes. If we notice those early we should already include fixes in the update PR. If we notice those later we should do follow ups. If we immediately notice that the fallout is very big than a revert can be considered especially if the update is not very important. 

But it is not your responsibility to fix **all** (last percent(s)) downstream packages. Otherwise no one could or would want to maintain core packages because it also draws in partly maintenance of 100 or sometimes 1000 downstream packages.

**Comment by risicle** [comment]
> because of single downstream failures

`arrow-cpp` has around 500 reverse dependencies of its own, so a "single failure" isn't really a single failure.

> But it is not your responsibility to fix all (last percent(s)) downstream packages. Otherwise no one could or would want to maintain core packages because it also draws in partly maintenance of 100 or sometimes 1000 downstream packages.

Core packages with this number of rebuilds should be merged to staging, not master. The lack of complete testing is compensated for by the opportunity to make fixes before breakage affects development of the >90% of PRs that target master.

(packages with ~500 rebuilds live in an uncomfortable hinterland of course)

**Comment by samuela** [comment]
> arrow-cpp has around 500 reverse dependencies of its own, so a "single failure" isn't really a single failure.

Agreed, if a PR targets master it shouldn't break any downstream packages. That should be uncontroversial.

**Comment by veprbl** [comment]
> > because of single downstream failures
> 
> `arrow-cpp` has around 500 reverse dependencies of its own, so a "single failure" isn't really a single failure.

Thankfully, it doesn't!

**Comment by samuela** [comment]
How many reverse transitive dependencies does grpc have? 500 seems realistic to me, but I don't know the exact number

**Comment by smancill** [comment]
What happens when a new version of gRPC breaks Arrow again, if https://github.com/NixOS/nixpkgs/pull/185329 is merged? What's the process to follow? gRPC would not be able to be updated until Arrow is updated? What if the Arrow update still doesn't work with that gRPC version?

**Comment by marsam** [comment]
>Could you submit a fix for arrow-cpp or revert this PR? In the future, please test all PRs with nixpkgs-review prior to merging.

Sorry about that. I think https://github.com/NixOS/nixpkgs/pull/185124 is going to be merged in the next couple of days, so I'll wait that instead of reverting this PR.

I maintain grpc because I use it in a hobby project, I'm not getting paid for it.
I know grpc is a core and I do try to fix any accidental breakage. See: 
https://github.com/NixOS/nixpkgs/pull/146580#issuecomment-974581000
https://github.com/NixOS/nixpkgs/pull/137055#issuecomment-917517665
but a couple of months ago my health deteriorated and I'm still recovering. 
Apologies if I'm a bit slow to respond.

**Comment by SuperSandro2000** [comment]
Don't worry. We'll figure it out and the next time it breaks we just take a look what can be done to fix it.

**Comment by veprbl** [comment]
@marsam Please take care of yourself first. For better or for worse, Nixpkgs can't ensure that for you.

**Comment by samuela** [comment]
@marsam Thanks for all the work that you do! I'm sorry to hear about your health, and best wishes for a swift recovery!

**Pull request merged**